Subject: [PATCH] gdb: Fix shift-negative-value error

* Fix below build error:
  arm-dis.c: In function 'print_insn_coprocessor':
  arm-dis.c:1953:20: error: left shift of negative value [-Werror=shift-negative-value]
         imm |= (-1 << 7);
                    ^~

* Original commit infomation as below:
  
  Fix for a "[-Werror=shift-negative-value]" compiler warning that is
  generated by "gdb-7.6/opcodes/arm-dis.c" when building crash with
  "make target=ARM64" on an x86_64 host with gcc-6.0.0.  Without the
  patch, the warning is treated as an error and the build is aborted.
 (anderson@redhat.com)
  More details as below:
  https://github.com/crash-utility/crash/commit/8df224099bcb9dd95cb2f8e16527c191f14c5d04

Upstream-Status: Backport

Signed-off-by: Mingli Yu <mingli.yu@windriver.com>
---
 gdb-7.3.1.patch | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/gdb-7.3.1.patch b/gdb-7.3.1.patch
index a338cf7..0c88d5b 100644
--- a/gdb-7.3.1.patch
+++ b/gdb-7.3.1.patch
@@ -1946,3 +1946,15 @@ index a1ef1c1..b05ed86 100644
  	   i++)
  	(*info->fprintf_func) (info->stream, "%s",
  			       prefix_name (all_prefixes[i], sizeflag));
+
+--- gdb-7.3.1/opcodes/arm-dis.c
++++ gdb-7.3.1/opcodes/arm-dis.c
+@@ -1950,7 +1950,7 @@ print_insn_coprocessor (bfd_vma pc,
+ 
+ 		    /* Is ``imm'' a negative number?  */
+ 		    if (imm & 0x40)
+-		      imm |= (-1 << 7);
++		      imm -= 0x80;
+ 
+ 		    func (stream, "%d", imm);
+ 		  }
-- 
2.8.2

